Serial node e2e jobs failing since 2026-07-22: kubelet panics at startup when PodLevelResources is disabled (PodLevelResourcesFixDefaulting)

### Which jobs are failing?

All serial node e2e jobs that run the CPU Manager or Memory Manager suites:

- ci-kubernetes-node-kubelet-containerd-resource-managers
- ci-node-crio-resource-managers
- ci-kubernetes-node-swap-ubuntu-serial
- ci-node-crio-swap-serial
- ci-containerd-node-e2e-serial-ec2
- ci-containerd-node-arm64-e2e-serial-ec2
- presubmits pull-kubernetes-node-kubelet-serial-cpu-manager and pull-kubernetes-node-kubelet-serial-cpu-manager-kubetest2 (for PRs rebased on master)

### Which tests are failing?

The first failure lands in the CPU Manager and Memory Manager suites that disable the PodLevelResources feature gate. After that, every later [Serial] test in the run fails in its BeforeAll or BeforeEach step with:

```
[FAILED] Failed to get successful response from /configz: context deadline exceeded
In [BeforeAll] at: k8s.io/kubernetes/test/e2e_node/kubeletconfig/kubeletconfig.go:186
```

Triage clusters (126 and 51 test names, same root cause, split only by ginkgo phase):

- https://storage.googleapis.com/k8s-triage/index.html?xjob=azure%7Ckops%7Ccrio%7Cautoscaling%7Ccapz%7Cci-kubernetes-verify%7Ccluster-api%7C-cos-%7Cfedora%7Cwindows%7Clocal-e2e%7Calpha-enabled-default%7Cci-gcp%7Cdocker-docker%7Cdocker-containerd%7Cci-minikube%7Ckvm-docker%7Cci-cloud-provider-gcp%7Cnpd-%7Cagentic-sandbox#fe142f553034eead5bbc
- https://storage.googleapis.com/k8s-triage/index.html?xjob=azure%7Ckops%7Ccrio%7Cautoscaling%7Ccapz%7Cci-kubernetes-verify%7Ccluster-api%7C-cos-%7Cfedora%7Cwindows%7Clocal-e2e%7Calpha-enabled-default%7Cci-gcp%7Cdocker-docker%7Cdocker-containerd%7Cci-minikube%7Ckvm-docker%7Cci-cloud-provider-gcp%7Cnpd-%7Cagentic-sandbox#2624ca6d5446537c85a5

### Since when has it been failing?

Since 2026-07-22. The last green build ran commit faba7e6a94df (merged 01:11 UTC). The first failing builds ran commit 39c618b5c30c (merged 04:09 UTC).

The clusters also contain builds from 2026-07-15 to 2026-07-17. That was the same failure mode, caused by #137150 and fixed by #140631. See the history section below.

### Reason for failure

The kubelet panics at startup while loading the config file the test wrote:

```
panic: failed to merge global and in-flight KubeletConfiguration while setting defaults,
error: PodLevelResourcesFixDefaulting is enabled, but depends on features that are
disabled: [PodLevelResources]
```

Mechanism:

1. #140514 (merged 2026-07-22 02:55 UTC) added the feature gate `PodLevelResourcesFixDefaulting`, enabled by default at Beta, with a declared dependency on `PodLevelResources` in `pkg/features/kube_features.go`.
2. `configureCPUManagerInKubelet` and `configureMemoryManagerInKubelet` in test/e2e_node set `PodLevelResources: false` in the kubelet config for some test variants, and do not disable the new gate.
3. `SetDefaults_KubeletConfiguration` (`pkg/kubelet/apis/config/v1beta1/defaults.go:72`) merges the config gates into a copy of the global gates with `SetFromMap`. The dependency check fails and the code panics.
4. systemd restarts the kubelet, it panics again on the same config, and /configz returns 503 through the API server proxy until the test times out. Every later Serial test in the run then fails the same way.

Evidence: [prow build](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-node-kubelet-containerd-resource-managers/2079792906901655552) and the [kubelet.log with the panic](https://storage.googleapis.com/kubernetes-ci-logs/logs/ci-kubernetes-node-kubelet-containerd-resource-managers/2079792906901655552/artifacts/n2d-standard-32-cos-125-19216-532-25-913c732d/kubelet.log).

History: this is the second time this exact pattern broke these jobs. #137150 added `PodLevelResourcesFixKubeletQOSClass` (also enabled by default, also depending on PodLevelResources) and broke the same jobs from 2026-07-15 to 2026-07-17. #140631 fixed the tests by disabling that gate by name, so the fix does not cover the new gate. The presubmit that runs these suites (pull-kubernetes-node-kubelet-serial-cpu-manager) is optional and did not run on #140514.

### Anything else we need to know?

This is not only a CI problem. Any kubelet config file that sets `featureGates: {PodLevelResources: false}` now crashes the kubelet at startup, and disabling a Beta gate is a normal thing for a cluster admin to do.

Two possible fixes:

- Short term, repeat #140631 for the new gate: also set `PodLevelResourcesFixDefaulting: false` next to the existing lines in `test/e2e_node/cpu_manager_test.go` and `test/e2e_node/memory_manager_test.go` when the test disables PodLevelResources.
- Longer term, stop the kubelet from panicking here. Either the dependency check should treat a gate that is only enabled by default as off when its dependency is off, or config loading should return an error instead of panicking. Without that, every future gate that depends on PodLevelResources and ships enabled by default will break these jobs again.

/kind failing-test
/kind regression
/sig node
/priority critical-urgent

cc @ndixita

--- a/test/e2e_node/cpu_manager_test.go
+++ b/test/e2e_node/cpu_manager_test.go
@@ -4294,9 +4294,10 @@ func configureCPUManagerInKubelet(oldCfg *kubeletconfig.KubeletConfiguration, ku
 	newCfg.FeatureGates["CPUManagerPolicyBetaOptions"] = kubeletArguments.enableCPUManagerOptions
 	newCfg.FeatureGates["CPUManagerPolicyAlphaOptions"] = kubeletArguments.enableCPUManagerOptions
 	newCfg.FeatureGates["PodLevelResources"] = kubeletArguments.enablePodLevelResources
-	// InPlacePodLevelResourcesVerticalScaling and PodLevelResourcesFixKubeletQOSClass are only supported when PodLevelResources is enabled
+	// InPlacePodLevelResourcesVerticalScaling, PodLevelResourcesFixDefaulting and PodLevelResourcesFixKubeletQOSClass are only supported when PodLevelResources is enabled
 	if !kubeletArguments.enablePodLevelResources {
 		newCfg.FeatureGates["InPlacePodLevelResourcesVerticalScaling"] = kubeletArguments.enablePodLevelResources
+		newCfg.FeatureGates["PodLevelResourcesFixDefaulting"] = kubeletArguments.enablePodLevelResources
 		newCfg.FeatureGates["PodLevelResourcesFixKubeletQOSClass"] = kubeletArguments.enablePodLevelResources
 	}
 	newCfg.FeatureGates["PodLevelResourceManagers"] = kubeletArguments.enablePodLevelResourceManagers

--- a/test/e2e_node/memory_manager_test.go
+++ b/test/e2e_node/memory_manager_test.go
@@ -808,9 +808,10 @@ func configureMemoryManagerInKubelet(oldCfg *kubeletconfig.KubeletConfiguration,
 	}
 
 	newCfg.FeatureGates["PodLevelResources"] = kubeletArguments.enablePodLevelResources
-	// InPlacePodLevelResourcesVerticalScaling and PodLevelResourcesFixKubeletQOSClass are only supported when PodLevelResources is enabled
+	// InPlacePodLevelResourcesVerticalScaling, PodLevelResourcesFixDefaulting and PodLevelResourcesFixKubeletQOSClass are only supported when PodLevelResources is enabled
 	if !kubeletArguments.enablePodLevelResources {
 		newCfg.FeatureGates["InPlacePodLevelResourcesVerticalScaling"] = kubeletArguments.enablePodLevelResources
+		newCfg.FeatureGates["PodLevelResourcesFixDefaulting"] = kubeletArguments.enablePodLevelResources
 		newCfg.FeatureGates["PodLevelResourcesFixKubeletQOSClass"] = kubeletArguments.enablePodLevelResources
 	}
 	newCfg.FeatureGates["PodLevelResourceManagers"] = kubeletArguments.enablePodLevelResourceManagers

